src/main/java/com/capgemini/wsb/fitnesstracker/training/api/dto/UpdateTrainingDto.java

```diff
@@ -0,0 +1,15 @@
+package com.capgemini.wsb.fitnesstracker.training.api.dto;
+import com.capgemini.wsb.fitnesstracker.user.api.User;
+import com.capgemini.wsb.fitnesstracker.training.internal.ActivityType;
+import java.util.Date;
+
+public record UpdateTrainingDto(
+        Long id,
+        Long userId,
+        User user,
+        Date startTime,
+        Date endTime,
+        ActivityType activityType,
+        double distance,
```

Comment:
Consider using object Double instead of primitive. On object you can do some operations that u can't do on primitive type e.g. u can handle nulls etc.